### PR TITLE
Correcting get_bool logic

### PR DESCRIPTION
**Pull request by lillekemiker**
Along with unit tests

### CODE REVIEW

File: tests/test_config_tree.py [hunk 1]
```diff
@@ -134,6 +135,46 @@ def test_getters_with_default(self):
         assert config_tree.get_config("config", {'b': 1}) == {'a': 5}
         assert config_tree.get_config("config-new", {'b': 1}) == {'b': 1}
 
+    def test_getter_type_conversion_string_to_bool(self):
```
Comment (dlsmith) [inline tests/test_config_tree.py]: Can you add a parser-based test that reads a boolean from the environment?

**Comment by dlsmith** [comment]
We should have a test that exercises the path through the environment, but otherwise looks good 👍

**Comment by lillekemiker** [comment]
@dlsmith : I just added three parser level unit tests. The last two might be crossing into ConfigTree territory, though. What do you think, should I remove them?

**Comment by dlsmith** [comment]
👍

**Comment by paulmelnikow** [comment]
@lillekemiker Can you open an upstream PR for this change?

**Comment by lillekemiker** [comment]
Done:
https://github.com/chimpler/pyhocon/pull/90

**Pull request merged**